clang/include/clang/AST/Type.h

```diff
@@ -251,15 +423,16 @@ class Qualifiers {
   }
 
   // Deserialize qualifiers from an opaque representation.
-  static Qualifiers fromOpaqueValue(unsigned opaque) {
+  static Qualifiers fromOpaqueValue(uint64_t Opaque) {
     Qualifiers Qs;
-    Qs.Mask = opaque;
+    Qs.Mask = uint32_t(Opaque);
```

Comment:
Is this OK or should we prefer smth like `Opaque & 0xFFFFFFFF` or even `Opaque & std::numeric_limits<uint32_t>::max()` (or smth else)?

Reply:
Either of those look better imo, or could even use `maskTrailingOnes` from `Support/MathExtras.h`

Reply:
Thanks, I've used numeric limits stuff since it allows `constexpr` (while `maskTrailingOnes` is more flexible for arbitrary bit count but does not have `constexpr` variant as far as I can see). See df81082d6f0210b02e72a3237ab7c03715d23488